src/elements/content-sidebar/DocGenSidebar/TagTree.tsx

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import './DocGenSidebar.scss';
+import { JsonData } from './types';
+
+interface TagTreeProps {
+    data: JsonData | string[];
+    level?: number;
+}
+
+const TagTree = ({ data, level = 0 }: TagTreeProps) => {
+    if (Array.isArray(data)) {
+        return null;
+    }
+
+    return (
```

Comment:
if it's an object, it should iterate recursively through it. If it's null - ts should protect use from null values at top level.

Reply:
So TS wont' actually protect you at runtime. Once the code gets transpiled it's pure JS and it doesn't know anything about types. Whenever I see recursion I always wanna make sure that we don't have any issues with not breaking out but based on what I see I think we should be fine even if data is null. It looks like it will just  throw an exception on Object.keys which will cause it to break out of the recursion.